pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -0,0 +1,530 @@
+							Namespace: dmsi.Spec.TargetSecretRef.Namespace,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return newSS
+
+}
+
+// delete snitches,secrets and syncset associated with the dmsi that has been deleted
+func (r *ReconcileDeadmansSnitchIntegration) deleteDMSI(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, clustDeploy *hivev1.ClusterDeployment, dmsc dmsclient.Client) error {
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", clustDeploy.Name, "cluster-deployment.Namespace:", clustDeploy.Namespace)
+	var DeadMansSnitchFinalizer string = "dms.managed.openshift.io/deadmanssnitch-" + dmsi.Name
+
+	// Delete the dms
+	logger.Info("Deleting the DMS from api.deadmanssnitch.com")
+	snitchName := clustDeploy.Spec.ClusterName + "." + clustDeploy.Spec.BaseDomain + "-" + dmsi.Spec.SnitchNamePostFix
+	snitches, err := dmsc.FindSnitchesByName(snitchName)
+	if err != nil {
+		return err
+	}
+	for _, s := range snitches {
+		delStatus, err := dmsc.Delete(s.Token)
+		if !delStatus || err != nil {
+			logger.Info("Failed to delete the DMS from api.deadmanssnitch.com")
+			return err
+		}
+		logger.Info("Deleted the DMS from api.deadmanssnitch.com")
+	}
+
+	// Delete the SyncSet
+	logger.Info("Deleting DMS SyncSet")
+	err = utils.DeleteSyncSet(clustDeploy.Name+"-"+dmsi.Spec.SnitchNamePostFix+config.RefSecretPostfix, clustDeploy.Namespace, r.client)
+	if err != nil {
+		logger.Error(err, "Error deleting SyncSet")
+		return err
+	}
+
+	// Delete the referenced secret
+	logger.Info("Deleting DMS referenced secret")
+	err = utils.DeleteRefSecret(clustDeploy.Name+"-"+dmsi.Spec.SnitchNamePostFix+config.RefSecretPostfix, clustDeploy.Namespace, r.client)
+	if err != nil {
+		logger.Error(err, "Error deleting secret")
+		return err
+	}
+
+	logger.Info("Deleting DMS finalizer from dmsi")
+	if utils.HasFinalizer(dmsi, DeadMansSnitchFinalizer) {
+		utils.DeleteFinalizer(dmsi, DeadMansSnitchFinalizer)
+		err = r.client.Update(context.TODO(), dmsi)
+		if err != nil {
+			logger.Error(err, "Error deleting Finalizer from dmsi")
+			return err
+		}
+
+		return nil
+	}
+	return nil
+
+}
+
+// delete snitches,secrets and syncset associated with the cluster deployment that has been deleted
+func (r *ReconcileDeadmansSnitchIntegration) deleteDMSClusterDeploy(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, clustDeploy *hivev1.ClusterDeployment, dmsc dmsclient.Client) error {
+
+	var DeadMansSnitchFinalizer string = "dms.managed.openshift.io/deadmanssnitch-" + dmsi.Name
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", clustDeploy.Name, "cluster-deployment.Namespace:", clustDeploy.Namespace)
+	// Delete the dms
+	logger.Info("Deleting the DMS from api.deadmanssnitch.com")
+	snitchName := clustDeploy.Spec.ClusterName + "." + clustDeploy.Spec.BaseDomain + "-" + dmsi.Spec.SnitchNamePostFix
```

Comment:
OSD doesn't currently append a post-fix so we need this to be an optional property.  I have a suggestion in the type for this.  If SnitchNamePostFix is not set / is empty don't append `-`.  This change needs to be done for SyncSet and Secret as well.

Reply:
@jewzaam is your intention that the OSD DMSI would reconcile the existing snitch & syncset & secret? I thought those were just going to be superseded by the new ones created here (with e.g. an `osd` suffix), and the old ones manually removed?

A problem I see with making the field optional is that if any other DMSI CR also doesn't specify it, then unexpected things will happen (or expected things won't).

Reply:
Fair point.  We'll have to be careful about migration then.  If we do not delete older snitches but do create new ones that get synced to the cluster we'll have every old snitch alert that the clusters have gone missing.